File: projects/grind-75/python/dynamic_programming/climbing_stairs.py
class Solution:
    @classmethod
    def climbStairs(self, n):
        if n == 1:
            return 1
        n_minus_2 = 1
        n_minus_1 = 2
        for _ in range(3, n + 1):
            tmp = n_minus_1
            n_minus_1 = n_minus_2 + n_minus_1
            n_minus_2 = tmp
        return n_minus_1

File: projects/grind-75/typescript/dynamic-programming/climbing-stairs.ts
export default function climbStairs(n: number): number {
  if (n === 1) {
    return 1;
  }
  let n_minus_2 = 1;
  let n_minus_1 = 2;
  for (let i = 3; i <= n; i++) {
    const tmp = n_minus_1;
    n_minus_1 = n_minus_2 + n_minus_1;
    n_minus_2 = tmp;
  }
  return n_minus_1;
}
